vm/src/exceptions.rs

```diff
@@ -862,13 +862,30 @@ fn key_error_str(exc: PyBaseExceptionRef, vm: &VirtualMachine) -> PyStrRef {
 
 fn os_error_str(exc: PyBaseExceptionRef, vm: &VirtualMachine) -> PyResult<PyStrRef> {
     let args = exc.args();
+    let obj = exc.as_object().to_owned();
+
     if args.as_slice().len() == 2 {
         // SAFETY: len() == 2 is checked so get_arg 1 or 2 won't panic
-        let s = format!(
-            "[Errno {}] {}",
-            exc.get_arg(0).unwrap().str(vm)?,
-            exc.get_arg(1).unwrap().str(vm)?
-        );
+        let errno = exc.get_arg(0).unwrap().str(vm)?;
+        let msg = exc.get_arg(1).unwrap().str(vm)?;
+
+        let s = match (
+            obj.clone().get_attr("filename", vm),
+            obj.get_attr("filename2", vm),
+        ) {
+            (Ok(filename), Ok(filename2)) => {
+                format!(
+                    "[Errno {}] {}: '{}' -> '{}'",
+                    errno,
+                    msg,
+                    filename.str(vm)?,
+                    filename2.str(vm)?
+                )
+            }
+            (Ok(filename), Err(_)) => format!("[Errno {}] {}: '{}'", errno, msg, filename.str(vm)?),
+            (Err(_), _) => format!("[Errno {}] {}", errno, msg),
+        };
```

Comment:
Pretty sure we can avoid one of the `get_attr` lookups here, i.e:

```suggestion
        let s = match obj.clone().get_attr("filename", vm) {
            Ok(filename) => match obj.get_attr("filename2", vm) {
                Ok(filename2) => format!(
                    "[Errno {}] {}: '{}' -> '{}'",
                    errno,
                    msg,
                    filename.str(vm)?,
                    filename2.str(vm)?
                ),
                _ => format!("[Errno {}] {}: '{}'", errno, msg, filename.str(vm)?),
            },
            _ => format!("[Errno {}] {}", errno, msg),
        };
```

Reply:
Its unfortunate we need to `getattr` here, though. CPython stores these in the struct so its super cheap to look-up. Either way, thats an issue for another PR.

Reply:
> 
> 
> Its unfortunate we need to `getattr` here, though. CPython stores these in the struct so its super cheap to look-up. Either way, thats an issue for another PR.

Totally agree with that. In the first place, I thought I can avoid the `get_attr` but later found out it is impossible.